scheduler/pkg/store/experiment/store.go

```diff
@@ -461,3 +471,22 @@ func (es *ExperimentStore) GetExperiments() ([]*Experiment, error) {
 	}
 	return foundExperiments, nil
 }
+
+func (es *ExperimentStore) cleanupDeletedExperiments() {
+	es.logger.Info("cleaning up deleted experiments")
+	for _, experiment := range es.experiments {
+		if experiment.Deleted {
+			es.mu.Lock()
+			defer es.mu.Unlock()
```

Comment:
I think the lock should be done once before the for loop? otherwise using `defer` here is potentially going to cause a deadlock.

Reply:
as mentioned above I think that this logic can also delete from badger db and not rely on ttl as well. just to make it simpler?